spec/support/http_handling_shared.rb

```diff
@@ -148,6 +148,18 @@
         expect(sockets_used.uniq.length).to eq(1)
       end
 
+      context "on a mixed state" do
+        it "re-opens the connection" do
+          first_socket = client.get("#{server.endpoint}/socket/1").body.to_s
```

Comment:
Yeah, just figured out that... Now that really makes more sense! :D But still dislike the names :D Although I can't come with better ones anyway :D